src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppDetailedView.tsx

```diff
@@ -0,0 +1,347 @@
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import space from 'app/styles/space';
+import {t, tct} from 'app/locale';
+import {addQueryParamsToExistingUrl} from 'app/utils/queryString';
+import {
+  installSentryApp,
+  uninstallSentryApp,
+} from 'app/actionCreators/sentryAppInstallations';
+import AsyncComponent from 'app/components/asyncComponent';
+import HookStore from 'app/stores/hookStore';
+import marked, {singleLineRenderer} from 'app/utils/marked';
+import InlineSvg from 'app/components/inlineSvg';
+import Tag from 'app/views/settings/components/tag';
+import {toPermissions} from 'app/utils/consolidatedScopes';
+import CircleIndicator from 'app/components/circleIndicator';
+import {Hooks} from 'app/types/hooks';
+import {
+  IntegrationFeature,
+  SentryApp,
+  Organization,
+  SentryAppInstallation,
+} from 'app/types';
+import withOrganization from 'app/utils/withOrganization';
+import {UninstallButton} from '../settings/organizationDeveloperSettings/sentryApplicationRow/installButtons';
+
+type State = {
+  sentryApp: SentryApp;
+  featureData: IntegrationFeature[];
+};
+
+type Props = {
+  organization: Organization;
+} & RouteComponentProps<{appSlug: string}, {}>;
+
+const defaultFeatureGateComponents = {
+  IntegrationFeatures: p =>
+    p.children({
+      disabled: false,
+      disabledReason: null,
+      ungatedFeatures: p.features,
+      gatedFeatureGroups: [],
+    }),
+  FeatureList: p => (
+    <ul>
+      {p.features.map((f, i) => (
+        <li key={i}>{f.description}</li>
+      ))}
+    </ul>
+  ),
+} as ReturnType<Hooks['integrations:feature-gates']>;
+
+class SentryAppDetailedView extends AsyncComponent<
+  Props & AsyncComponent['props'],
+  State & AsyncComponent['state']
+> {
+  getEndpoints(): ([string, string, any] | [string, string])[] {
+    const {
+      organization,
+      params: {appSlug},
+    } = this.props;
+    const baseEndpoints: ([string, string, any] | [string, string])[] = [
+      ['sentryApp', `/sentry-apps/${appSlug}/`],
+      ['featureData', `/sentry-apps/${appSlug}/features/`],
+      ['appInstalls', `/organizations/${organization.slug}/sentry-app-installations/`],
+    ];
+
+    return baseEndpoints;
+  }
+
+  featureTags(features: IntegrationFeature[]) {
+    return features.map(feature => {
+      const feat = feature.featureGate.replace(/integrations/g, '');
+      return <StyledTag key={feat}>{feat.replace(/-/g, ' ')}</StyledTag>;
+    });
+  }
+
+  get permissions() {
+    return toPermissions(this.state.sentryApp.scopes);
+  }
+
+  isInstalled = () => {
+    return this.state.appInstalls.find(i => i.app.slug === this.state.sentryApp.slug);
+  };
+
+  redirectUser = (install: SentryAppInstallation) => {
+    const {organization} = this.props;
+    const {sentryApp} = this.state;
+    if (!sentryApp.redirectUrl) {
+      addSuccessMessage(t(`${sentryApp.slug} successfully installed.`));
+      this.setState({appInstalls: [install, ...this.state.appInstalls]});
```

Comment:
@NisanthanNanthakumar I think this would be better to put in `handleInstall` since it relates to the installation succeeding. It was a bit different before when we had the callback.